02-Generics/Generics/Generics.cs

```diff
@@ -92,9 +107,26 @@ public static class ArrayExtentions {
 		///   }
 		/// </example>
 		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) {
-			// TODO :SortTupleArray<T1, T2, T3>
-			// HINT : Add required constraints to generic types
-		}
+            // TODO :SortTupleArray<T1, T2, T3>
+            // HINT : Add required constraints to generic types
+
+            if (!(sortedColumn >= 0 && sortedColumn < 3)) throw new IndexOutOfRangeException();
+
+            var buff = new Dictionary<int, Func<Tuple<T1, T2, T3>, IComparable>>()
+                {
+                    {0,(arg)=>{return (IComparable) arg.Item1; } },
+                    {1,(arg)=>{return (IComparable) arg.Item2; } },
+                    {2,(arg)=>{return (IComparable) arg.Item3; } }
```

Comment:
You don't need to cast the item, just add  required constraints to generic types.